### PR TITLE
Merge from master (September 2018)

### PR DESCRIPTION
**Pull request by pquentin**
As with all merges, this cannot really be reviewed usefully, unfortunately. I'm quite confident in the merge.

### CODE REVIEW

File: test/with_dummyserver/test_socketlevel.py [hunk 1]
```diff
@@ -1442,14 +1442,14 @@ def _test_broken_header_parsing(self, headers):
 
     def test_header_without_name(self):
         self._test_broken_header_parsing([
-            b': Value\r\n',
-            b'Another: Header\r\n',
+            b': Value',
+            b'Another: Header',
```
Comment (njsmith) [inline test/with_dummyserver/test_socketlevel.py]: These `\r\n` changes don't make much sense to me. Should it also have switched to `b'\r\n'.join(headers)` up above?

File: test/with_dummyserver/test_socketlevel.py [hunk 2]
```diff
@@ -1428,12 +1428,12 @@ def socket_handler(listener):
...
 class TestBrokenHeaders(SocketDummyServerTestCase):
 
-    def _test_broken_header_parsing(self, headers):
+    def _test_broken_header_parsing(self, headers, unparsed_data_check=None):
         self.start_response_handler((
            b'HTTP/1.1 200 OK\r\n'
            b'Content-Length: 0\r\n'
            b'Content-type: text/plain\r\n'
-           ) + b''.join(headers) + b'\r\n'
+           ) + b''.join(headers) + b'\r\n\r\n'
```
Comment (njsmith) [inline test/with_dummyserver/test_socketlevel.py]: Ok yeah if you look at https://github.com/python-trio/urllib3/commit/285889d57f8bb00116617dc654124a4c7d223fed then it has `b'\r\n'` here. I'm not sure how or why we diverged, but the changes below are assuming it's `b'\r\n'`

Comment (pquentin) [reply test/with_dummyserver/test_socketlevel.py]: For reference, I think you meant https://github.com/python-trio/urllib3/pull/95/commits/f4efcca559cc753f878e0a73e28dd6e795d0d052

File: test/with_dummyserver/test_socketlevel.py (hunk 1 above)
Comment (pquentin) [reply test/with_dummyserver/test_socketlevel.py]: Okay, so the [initial commit in the master branch](https://github.com/urllib3/urllib3/commit/5c2cf76ddd2d42aa1906aef5eb08c6891a9b2614) had those extra \r\n, which was a mistake. This was fixed twice:

 * [in the v2 branch](https://github.com/urllib3/urllib3/commit/f220a99eb2a1b1bb3d9ae30b58904c318fce63db) by switching to `b''.join()`
 * [in the master branch (the commit I just merged)](https://github.com/urllib3/urllib3/commit/f4efcca559cc753f878e0a73e28dd6e795d0d052) by fixing the individual test cases and keeping `b'\r\n'.join()`.

So I just pushed a new commit that follows the approach of the master branch, to help future merges. This should make more sense!

**Comment by pquentin** [comment]
@njsmith Please take a look.

**Comment by pquentin** [comment]
@njsmith Thank you for the review!

**Pull request merged**